### PR TITLE
Feature/condo/sberdoma 128/auth register etc

### PR DESCRIPTION
**Pull request by dkoviazin**
Preview

### CODE REVIEW

File: apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,246 @@
+import { PageHeader as AntPageHeader, PageHeaderProps } from 'antd'
+import React from 'react'
+import Router from 'next/router'
+import { FormattedMessage } from 'react-intl'
+import { Layout } from 'antd'
+const { Footer, Content } = Layout
+import { Logo } from '@condo/domains/common/components/Logo'
+import { SUPPORT_EMAIL, SUPPORT_PHONE } from '@condo/domains/common/constants/requisites'
+
+const LINK_STYLE = { color: '#389E0D' }
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx:10]: move to domains/common/consts

File: apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+const PageContent: React.FC = ({ children }) => {
+    return (
+        <Content style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: 'calc(100vh - 200px)', minWidth: '500px' }}>
+            {children}
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx:15]: wrap children inot div and set div margin: auto

File: apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        </Content>
+    )
+}
+
+const PageFooter: React.FC = () => {
+    return (
+        <Footer style={{ color: 'gray', backgroundColor: 'white', fontSize: '12px', lineHeight: '20px' }}>
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx:22]: from constns

File: apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+
+interface IAuthLayoutProps {
+    headerAction: React.ReactElement
+}
+
+const AuthLayout: React.FC<IAuthLayoutProps> = ({ children, headerAction }) => {
+    return (
+        <Layout
+            style={{ background: 'white' }}
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx:44]: from consts

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            type: 'type AuthenticateUserWithPhoneAndPasswordOutput { item: User }',
+        },
+    ],
+    mutations: [
+        {
+            access: true,
+            schema: 'authenticateUserWithPhoneAndPassword(data: AuthenticateUserWithPhoneAndPasswordInput!): AuthenticateUserWithPhoneAndPasswordOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                await AuthenticateUserWithPhoneAndPasswordService.emit('beforeAuthenticateUserWithPhoneAndPassword', {
+                    parent, args, context, info, extra,
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/AuthPassword.js:22]: remove hook

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                })
+                const { phone: inputPhone, password } = info.variableValues
+                const phone = inputPhone.replace(PHONE_CLEAR_REGEXP, '')
+                console.log('Clear phone ', inputPhone, phone)
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/AuthPassword.js:26]: remove log

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                          objs: allUsers(where: { phone: $phone }) {
+                            id
+                            phone
+                          }
+                        }
+                    `,
+                    variables: { phone },
+                })
+
+                if (findErrors || !findData.objs || findData.objs.length !== 1 || !findData.objs[0].id) {
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/AuthPassword.js:41]: !findData.objs[0].id can throw exceptions

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+                    const msg = '[notfound.error] Unable to find user. Try to register'
+                    console.error(msg, findErrors)
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/AuthPassword.js:43]: remove log

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+                const userData = findData.objs[0]
+
+                const { keystone } = await getSchemaCtx(AuthenticateUserWithPhoneAndPasswordService)  
+
+                const { auth:{ User: { password:PasswordStrategy } } } = keystone
+                
+                const { success, message } = await PasswordStrategy.validate({ phone, password })
+                
+                if (!success) {
+                    console.error(message, 'Validate password')
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/AuthPassword.js:56]: remove log

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+                    throw new Error(message)
+                }
+                await context.startAuthedSession({ item: userData, list: keystone.lists['User'] })
+                const result = {
+                    item: await getById('User', userData.id),
+                }
+                await AuthenticateUserWithPhoneAndPasswordService.emit('afterAuthenticateUserWithFirebaseIdToken', {
+                    parent, args, context, info, extra, result,
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/AuthPassword.js:64]: remove hook

File: apps/condo/domains/user/schema/ForgotPassword.js [hunk 11]
```diff
@@ -0,0 +1,286 @@
...
+                    throw new Error('[unknown-user]: Unable to find user when trying to start password recovery')
+                }
+
+                if (userData.allUsers.length !== 1) {
+                    throw new Error('[unknown-user]: Unable to find exact one user to start password recovery')
+                }
+
+                const userId = userData.allUsers[0].id
+                const user = userData.allUsers[0]
+                console.log('User found ', userId, user)
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/ForgotPassword.js:108]: remove log

File: apps/condo/domains/user/schema/ForgotPassword.js [hunk 12]
```diff
@@ -0,0 +1,286 @@
...
+                const { errors: passwordError } = await context.executeGraphQL({
+                    context: context.createContext({ skipAccessControl: true }),
+                    query: `
+                        mutation UpdateUserPassword($user: ID!, $password: String!) {
+                          updateUser(id: $user, data: { password: $password }) {
+                            id
+                          }
+                        }
+                    `,
+                    variables: { user, password },
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/ForgotPassword.js:258]: add pass validation

File: apps/condo/lang/ru.json [hunk 13]
```diff
@@ -179,6 +176,48 @@
...
+  "pages.auth.register.field.Phone": "Телефон", 
+  "pages.auth.register.info.UserAgreement": "Нажимая на кнопку «Зарегистрироваться» вы принимаете {link}", 
+  "pages.auth.register.info.UserAgreementFileName": "условия политики конфиденциальности", 
+  "pages.auth.register.ChangePhoneNumber": "Изменить номер", 
+  
+  
+  
+
+
+  "pages.auth.FooterText": "Проблемы с регистрацией или входом? Напишите на {email} или позвоните по телефону {phone}",
```
Comment (Dimitreee) [inline apps/condo/lang/ru.json:220]: add translation to en.json

File: apps/condo/pages/auth/auth.tsx [hunk 14]
```diff
@@ -0,0 +1,35 @@
+
```
Comment (Dimitreee) [inline apps/condo/pages/auth/auth.tsx:1]: remove file

File: apps/condo/pages/auth/change-password.tsx [hunk 15]
```diff
@@ -1,152 +1,166 @@
...
+const INPUT_STYLE = { width: '273px' }
 
 const CHANGE_PASSWORD_WITH_TOKEN_MUTATION = gql`
     mutation changePasswordWithToken($token: String!, $password: String!) {
         status: changePasswordWithToken(token: $token, password: $password)
     }
 `
 
-const ChangePasswordForm = () => {
+const ChangePasswordPage = (): React.ReactElement => {
```
Comment (Dimitreee) [inline apps/condo/pages/auth/change-password.tsx:25]: React.FC

Comment (Dimitreee) [reply apps/condo/pages/auth/change-password.tsx:25]:
```suggestion
const ChangePasswordPage:React.FC = () => {
```

File: apps/condo/pages/auth/change-password.tsx [hunk 16]
```diff
@@ -1,152 +1,166 @@
...
-    const GoToLoginMsg = intl.formatMessage({ id: 'GoToLogin' })
-    const PasswordWasChangedMsg = intl.formatMessage({ id: 'pages.auth.PasswordWasChanged' })
-    const PasswordWasChangedDescriptionMsg = intl.formatMessage({ id: 'pages.auth.PasswordWasChangedDescription' })
+    const [changePassword] = useMutation(CHANGE_PASSWORD_WITH_TOKEN_MUTATION)
+
+    const intl = useIntl()
+    const SaveMsg = intl.formatMessage({ id: 'Save' })
+    const PasswordMsg = intl.formatMessage({ id: 'pages.auth.signin.field.Password' })
+    const ResetTitle = intl.formatMessage({ id: 'pages.auth.ResetTitle' })
+    const CreateNewPassword = intl.formatMessage({ id: 'pages.auth.reset.CreateNewPasswordMsg' })
```
Comment (Dimitreee) [inline apps/condo/pages/auth/change-password.tsx:36]: Message

File: apps/condo/pages/auth/change-password.tsx [hunk 17]
```diff
@@ -1,152 +1,166 @@
...
-        <Typography.Title css={css`text-align: center;`} level={2}>{ChangePasswordTitleMsg}</Typography.Title>
-        <ChangePasswordForm />
-    </>)
+    const RegisterTitle = intl.formatMessage({ id: 'pages.auth.Register' })
+    return (
+        <Button
+            key='submit'
+            onClick={() => Router.push('/auth/register')}
+            type='sberPrimary'
+            secondary={true}
```
Comment (Dimitreee) [inline apps/condo/pages/auth/change-password.tsx:154]: add size prop

File: apps/condo/pages/auth/forgot.tsx [hunk 18]
```diff
@@ -1,124 +1,143 @@
...
+                </Typography.Paragraph>    
+            </div>
+        )    
+    }
 
-    const onFinish = values => {
+    const onSubmit = values => {
         if (values.email) values.email = values.email.toLowerCase()
         setIsLoading(true)
+        console.log('Run mutation')
```
Comment (Dimitreee) [inline apps/condo/pages/auth/forgot.tsx:68]: remove log

File: apps/condo/pages/auth/forgot.tsx [hunk 19]
```diff
@@ -1,124 +1,143 @@
...
                 setIsLoading(false)
+                setIsSuccessMessage(true)
             },
+            finally: () => setIsLoading(false),
             intl,
             form,
             ErrorToFormFieldMsgMapping,
+        }).catch(err => {
+            setIsLoading(false)
+            console.log('Run mutation err ', err )
```
Comment (Dimitreee) [inline apps/condo/pages/auth/forgot.tsx:82]: remove log

File: apps/condo/pages/auth/forgot.tsx [hunk 20]
```diff
@@ -1,124 +1,143 @@
...
-        <ForgotForm />
-    </>)
+    const RegisterTitle = intl.formatMessage({ id: 'pages.auth.Register' })
+    return (
+        <Button
+            key='submit'
+            onClick={() => Router.push('/auth/register')}
+            type='sberPrimary'
+            secondary={true}
+            style={{ fontSize: '16px', lineHeight: '24px' }}
```
Comment (Dimitreee) [inline apps/condo/pages/auth/forgot.tsx:132]: try to use Size prop

File: apps/condo/pages/auth/register.tsx [hunk 21]
```diff
@@ -1,473 +1,297 @@
...
+import { formatPhone } from '@condo/domains/common/utils/helpers'
+import MaskedInput from 'antd-mask-input'
 
-            setUser(user)
-        })
-    }, [])
+const INPUT_STYLE = { width: '273px' }
+const POLICY_LOCATION = '/policy.pdf'
+const TEST_PHONE = '+79226052265'
+const LINK_STYLE = { color: '#389E0D' }
```
Comment (Dimitreee) [inline apps/condo/pages/auth/register.tsx:15]: use form consts

File: apps/condo/pages/auth/register.tsx [hunk 22]
```diff
@@ -1,473 +1,297 @@
...
+import { FormattedMessage } from 'react-intl'
+import { formatPhone } from '@condo/domains/common/utils/helpers'
+import MaskedInput from 'antd-mask-input'
 
-            setUser(user)
-        })
-    }, [])
+const INPUT_STYLE = { width: '273px' }
+const POLICY_LOCATION = '/policy.pdf'
+const TEST_PHONE = '+79226052265'
```
Comment (Dimitreee) [inline apps/condo/pages/auth/register.tsx:14]: remove test data

File: apps/condo/pages/auth/register.tsx [hunk 23]
```diff
@@ -1,473 +1,297 @@
...
+import React, { useEffect, useState } from 'react'
+import { FormattedMessage } from 'react-intl'
+import { formatPhone } from '@condo/domains/common/utils/helpers'
+import MaskedInput from 'antd-mask-input'
 
-            setUser(user)
-        })
-    }, [])
+const INPUT_STYLE = { width: '273px' }
+const POLICY_LOCATION = '/policy.pdf'
```
Comment (Dimitreee) [inline apps/condo/pages/auth/register.tsx:13]: discuss with PRODUCT

File: apps/condo/pages/auth/signin.tsx [hunk 24]
```diff
@@ -1,170 +1,172 @@
...
+import { useMutation } from '@core/next/apollo'
+import { useAuth } from '@core/next/auth'
 
-import { Auth, PhoneAuthForm } from './register'
 
-const SIGNIN_BY_FIREBASE_ID_TOKEN_MUTATION = gql`
-    mutation authenticateUserWithFirebaseIdToken ($token: String!) {
-        obj: authenticateUserWithFirebaseIdToken(data: { firebaseIdToken: $token }) {
+
+const SIGNIN_BY_PHONE_AND_PASSWORD_MUTATION = gql`
```
Comment (Dimitreee) [inline apps/condo/pages/auth/signin.tsx:18]: move to gql

File: apps/condo/pages/auth/signin.tsx [hunk 25]
```diff
@@ -1,170 +1,172 @@
...
             }
         }
     }
-`
+` 
 
-const SignInForm = ({ firebaseUser, children, ExtraErrorToFormFieldMsgMapping = {} }) => {
-    const [form] = Form.useForm()
+const INPUT_STYLE = { minWidth: '273px' }
+const LINK_STYLE = { color: '#389E0D' }
```
Comment (Dimitreee) [inline apps/condo/pages/auth/signin.tsx:29]: use const colors

File: apps/condo/pages/auth/signin.tsx [hunk 26]
```diff
@@ -1,170 +1,172 @@
...
-                token: firebaseUser.token,
-            },
+            mutation: signinByPhoneAndPassword,
+            variables: values,
             onCompleted: () => {
                 refetch()
-                if (initialValues.next) Router.push(initialValues.next)
-                else Router.push('/')
+                Router.push(next ? next : '/')
+                console.log('COMPLETE ! LOGIN ')
```
Comment (Dimitreee) [inline apps/condo/pages/auth/signin.tsx:84]: remove logging

File: apps/condo/pages/auth/signin.tsx [hunk 27]
```diff
@@ -1,170 +1,172 @@
...
             },
-            onError: (e) => {
-                console.error(e.friendlyDescription)
-            },
             intl,
             form,
             ErrorToFormFieldMsgMapping,
-            OnCompletedMsg: LoggedInMsg,
+        }).catch(error => {
+            console.log('Mutation error: ', error)
```
Comment (Dimitreee) [inline apps/condo/pages/auth/signin.tsx:93]: remove logging

File: apps/condo/pages/auth/signin.tsx [hunk 28]
```diff
@@ -1,170 +1,172 @@
...
+            <div>
+                <Typography.Title>{SignInTitleMsg}</Typography.Title>
+                <LoginForm />
+            </div>
+        </>
+    )
+}
 
+
+const LoginForm = (): React.ReactElement => {
```
Comment (Dimitreee) [inline apps/condo/pages/auth/signin.tsx:49]: rename form

File: apps/condo/pages/auth/signin.tsx [hunk 29]
```diff
@@ -1,170 +1,172 @@
...
-        </SignInForm>
-    }} />
+    const RegisterTitle = intl.formatMessage({ id: 'pages.auth.Register' })
+    return (
+        <Button
+            key='submit'
+            onClick={() => Router.push('/auth/register')}
+            type='sberPrimary'
+            secondary={true}
+            style={{ fontSize: '16px', lineHeight: '24px' }}
```
Comment (Dimitreee) [inline apps/condo/pages/auth/signin.tsx:161]: look at size

File: apps/condo/pages/authold/forgot.tsx [hunk 30]
```diff
@@ -0,0 +1,124 @@
...
+import Router from 'next/router'
+import { useIntl } from '@core/next/intl'
+import { gql } from 'graphql-tag'
+import { useMutation } from '@core/next/apollo'
+
+import { TopMenuOnlyLayout } from '@condo/domains/common/components/containers/BaseLayout'
+import { getQueryParams } from '@condo/domains/common/utils/url.utils'
+import { runMutation } from '@condo/domains/common/utils/mutations.utils'
+
+const START_PASSWORD_RECOVERY_MUTATION = gql`
```
Comment (Dimitreee) [inline apps/condo/pages/authold/forgot.tsx:16]: move to gql

File: apps/condo/pages/authold/forgot.tsx [hunk 31]
```diff
@@ -0,0 +1,124 @@
+// @ts-nocheck
```
Comment (Dimitreee) [inline apps/condo/pages/authold/forgot.tsx:1]: remove old backup forgot page

File: apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx:15]: Done - wrapped children into div with min. width - to remove all widths in auth pages

Parent container still needs height attribute. margin auto works the same as 
```css
justify-content: 'center';
align-items: 'center';
```
Skipped for now and added TODO - will ask HTML professional for a solution

File: apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx:22]: Moved to constants, used color from colors constant.

File: apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx (hunk 4 above)
Comment (dkoviazin) [reply apps/condo/domains/common/components/containers/BaseLayout/AuthLayout.tsx:44]: Done

File: apps/condo/domains/user/schema/AuthPassword.js (hunk 5 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/AuthPassword.js:22]: Done

File: apps/condo/domains/user/schema/AuthPassword.js (hunk 6 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/AuthPassword.js:26]: Done

File: apps/condo/domains/user/schema/AuthPassword.js (hunk 7 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/AuthPassword.js:41]: Fixed

File: apps/condo/domains/user/schema/AuthPassword.js (hunk 8 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/AuthPassword.js:43]: Done

File: apps/condo/domains/user/schema/ForgotPassword.js (hunk 12 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/ForgotPassword.js:258]: Done

File: apps/condo/pages/auth/auth.tsx (hunk 14 above)
Comment (dkoviazin) [reply apps/condo/pages/auth/auth.tsx:1]: Done

File: apps/condo/pages/auth/register.tsx (hunk 23 above)
Comment (dkoviazin) [reply apps/condo/pages/auth/register.tsx:13]: now it's a link to an empty file. I Will put the good file to the repository later

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 32]
```diff
@@ -0,0 +1,74 @@
...
+            schema: 'authenticateUserWithPhoneAndPassword(data: AuthenticateUserWithPhoneAndPasswordInput!): AuthenticateUserWithPhoneAndPasswordOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                await AuthenticateUserWithPhoneAndPasswordService.emit('beforeAuthenticateUserWithPhoneAndPassword', {
+                    parent, args, context, info, extra,
+                })
+                const { phone: inputPhone, password } = info.variableValues
+                const phone = inputPhone.replace(PHONE_CLEAR_REGEXP, '')
+                console.log('Clear phone ', inputPhone, phone)
+
+                const { errors: findErrors, data: findData } = await context.executeGraphQL({
```
Comment (pahaz) [inline apps/condo/domains/user/schema/AuthPassword.js:28]:
```
const { User } = require('@condo/domains/user/utils/serverSchema')
...

                const users = await User.getAll(context, { phone: phone_number })
                if (users.length !== 1) {
                    const msg = '[notfound.error] Unable to find user. Try to register'
                    console.error(msg, findErrors)
                    throw new Error(msg)
                }
...
```

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 33]
```diff
@@ -0,0 +1,74 @@
...
+                    const msg = '[notfound.error] Unable to find user. Try to register'
+                    console.error(msg, findErrors)
+                    throw new Error(msg)
+                }
+
+                const userData = findData.objs[0]
+
+                const { keystone } = await getSchemaCtx(AuthenticateUserWithPhoneAndPasswordService)  
+
+                const { auth:{ User: { password:PasswordStrategy } } } = keystone
```
Comment (pahaz) [inline apps/condo/domains/user/schema/AuthPassword.js:51]:
```suggestion
                const { auth: { User: { password: PasswordStrategy } } } = keystone
```

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 34]
```diff
@@ -0,0 +1,74 @@
...
+                
+                if (!success) {
+                    console.error(message, 'Validate password')
+                    throw new Error(message)
+                }
+                await context.startAuthedSession({ item: userData, list: keystone.lists['User'] })
+                const result = {
+                    item: await getById('User', userData.id),
+                }
+                await AuthenticateUserWithPhoneAndPasswordService.emit('afterAuthenticateUserWithFirebaseIdToken', {
```
Comment (pahaz) [inline apps/condo/domains/user/schema/AuthPassword.js:63]: you can drop it! we don't needed emits anymore

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 35]
```diff
@@ -0,0 +1,74 @@
...
+            access: true,
+            type: 'type AuthenticateUserWithPhoneAndPasswordOutput { item: User }',
+        },
+    ],
+    mutations: [
+        {
+            access: true,
+            schema: 'authenticateUserWithPhoneAndPassword(data: AuthenticateUserWithPhoneAndPasswordInput!): AuthenticateUserWithPhoneAndPasswordOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                await AuthenticateUserWithPhoneAndPasswordService.emit('beforeAuthenticateUserWithPhoneAndPassword', {
```
Comment (pahaz) [inline apps/condo/domains/user/schema/AuthPassword.js:21]: drop it please 5)

File: apps/condo/domains/user/schema/ForgotPassword.spec.js [hunk 36]
```diff
@@ -0,0 +1,105 @@
+
```
Comment (pahaz) [inline apps/condo/domains/user/schema/ForgotPassword.spec.js:1]: ForgotPassword.test.js

File: apps/condo/pages/auth/signin.tsx [hunk 37]
```diff
@@ -1,170 +1,172 @@
...
     const ErrorToFormFieldMsgMapping = {
-        [WRONG_EMAIL_ERROR]: {
-            name: 'email',
-            errors: [EmailIsNoFoundMsg],
+        '[notfound.error]': {
+            name: 'phone',
+            errors: [UserNotFound],
         },
-        [WRONG_PASSWORD_ERROR]: {
+        '[passwordAuth:secret:mismatch]': {
```
Comment (pahaz) [inline apps/condo/pages/auth/signin.tsx:70]: extract the error const inside user/constants

File: apps/condo/domains/user/schema/ForgotPassword.spec.js (hunk 36 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/ForgotPassword.spec.js:1]: Done

File: apps/condo/domains/user/schema/AuthPassword.js (hunk 35 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/AuthPassword.js:21]: Done

File: apps/condo/domains/user/schema/AuthPassword.js (hunk 33 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/AuthPassword.js:51]: Done

File: apps/condo/domains/user/schema/AuthPassword.js (hunk 32 above)
Comment (dkoviazin) [reply apps/condo/domains/user/schema/AuthPassword.js:28]: Done. Thx

File: apps/condo/pages/auth/change-password.tsx (hunk 15 above)
Comment (dkoviazin) [reply apps/condo/pages/auth/change-password.tsx:25]: SKipped for now - add TODO - need to fix typescript validation

File: apps/condo/lang/ru.json (hunk 13 above)
Comment (dkoviazin) [reply apps/condo/lang/ru.json:206]: Done

File: apps/condo/pages/auth/signin.tsx (hunk 37 above)
Comment (dkoviazin) [reply apps/condo/pages/auth/signin.tsx:60]: Skipped. Need to talk about error constants

File: apps/condo/domains/user/schema/ForgotPassword.js [hunk 38]
```diff
@@ -0,0 +1,289 @@
...
+    },
+})
+
+const ForgotPasswordService = new GQLCustomSchema('ForgotPasswordService', {
+    mutations: [
+        {
+            access: true,
+            schema: 'startPasswordRecovery(email: String!): String',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                await ForgotPasswordService.emit('beforeStartPasswordRecovery', {
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/ForgotPassword.js:76]: remove emits

File: apps/condo/domains/user/schema/ForgotPassword.js [hunk 39]
```diff
@@ -0,0 +1,289 @@
...
+                    sender: BOT_EMAIL,
+                })
+
+
+                // hook for send mail!
+                // SendMail `${SERVER_URL}/auth/change-password?token=${ForgotPasswordAction.token}` User.email
+                //                 
+                // 
+
+                // console.log('ForgotPasswordService AFTER', `${SERVER_URL}/auth/change-password?token=${ForgotPasswordAction.token}`)
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/ForgotPassword.js:196]: remove comments

File: apps/condo/domains/user/schema/ForgotPassword.js [hunk 40]
```diff
@@ -0,0 +1,289 @@
...
+                    meta: {
+                        token,
+                        dv: 1,
+                    },
+                    sender: BOT_EMAIL,
+                })
+
+
+                // hook for send mail!
+                // SendMail `${SERVER_URL}/auth/change-password?token=${ForgotPasswordAction.token}` User.email
```
Comment (Dimitreee) [inline apps/condo/domains/user/schema/ForgotPassword.js:192]: remove comments

**Review by Dimitreee — APPROVED** [review]

File: apps/condo/package.json [hunk 41]
```diff
@@ -44,7 +44,7 @@
     "@zeit/next-less": "^1.0.1",
     "ajv": "^8.2.0",
     "antd": "^4.8.2",
-    "antd-mask-input": "^0.1.14",
+    "antd-mask-input": "^0.1.15",
```
[the remark this answers is not in the document]
Comment (dkoviazin) [reply apps/condo/package.json:47]: It's not in the lock file and it's added by me for auth forms - it is not  used anywhere else

File: apps/condo/domains/user/schema/ForgotPassword.test.js [hunk 42]
```diff
@@ -0,0 +1,106 @@
...
+            'path': ['objs'],
+        })
+        expect(data).toEqual({ objs: null })
+    })
+
+    test('reset forgotten password', async () => {
+        const [user, userAttrs] = await createTestUser(await makeLoggedInAdminClient())
+        const client = await makeClient()
+        const adm = await makeLoggedInAdminClient()
+        const res1 = await client.mutate(START_PASSWORD_RECOVERY_MUTATION, { email: userAttrs.email })
```
Comment (pahaz) [inline apps/condo/domains/user/schema/ForgotPassword.test.js:72]:
```suggestion
        const admin = await makeLoggedInAdminClient()
        const [user, userAttrs] = await createTestUser(admin)
        const client = await makeClient()

        const res1 = await client.mutate(START_PASSWORD_RECOVERY_MUTATION, { email: userAttrs.email })
```

File: apps/condo/domains/user/schema/ForgotPassword.test.js [hunk 43]
```diff
@@ -0,0 +1,106 @@
...
+    })
+
+    test('start recovery for unknown email', async () => {
+        const client = await makeClient()
+        const res1 = await client.mutate(START_PASSWORD_RECOVERY_MUTATION, { email: `r${Math.random()}@emample.com` })
+        expect(JSON.stringify(res1.errors)).toEqual(expect.stringMatching('unknown-user'))
+    })
+
+    // zuch(todo): Ask why we need this. Now we have validation for 7 symbols length in mutation
+    test.skip('change password to empty', async () => {
```
Comment (pahaz) [inline apps/condo/domains/user/schema/ForgotPassword.test.js:97]: please check it

File: apps/condo/domains/user/schema/index.js [hunk 44]
```diff
@@ -4,11 +4,16 @@
  */
 
 const { User } = require('./User')
-const { AuthenticateUserWithFirebaseIdTokenService } = require('./Auth')
+const { AuthenticateUserWithFirebaseIdTokenService } = require('./AuthFirebase')
+const { AuthenticateUserWithPhoneAndPasswordService } = require('./AuthPassword')
```
Comment (pahaz) [inline apps/condo/domains/user/schema/index.js:8]: AuthenticateUserWithPhoneAndPasswordService.js

File: apps/condo/pages/auth/register-new.tsx [hunk 45]
```diff
@@ -0,0 +1,299 @@
...
+                    </Col>
+                    <Col span={24}>
+                        <Typography.Paragraph style={{ textAlign: 'left', fontSize: '12px', marginTop: '12px', lineHeight: '20px' }}>
+                        
+                            <FormattedMessage
+                                id='pages.auth.register.info.UserAgreement'
+                                values={{
+                                    link: <a style={LINK_STYLE} target='_blank' href={POLICY_LOCATION} rel="noreferrer">{UserAgreementFileName}</a>,
+                                }}
+                            ></FormattedMessage>
```
Comment (pahaz) [inline apps/condo/pages/auth/register-new.tsx:85]:
```suggestion
                            />
```

File: apps/condo/pages/auth/register-new.tsx [hunk 46]
```diff
@@ -0,0 +1,299 @@
...
+                        </Form.Item>
+                    </Col>
+                    <Col span={24}>
+                        <Form.Item
+                            name="email"
+                            label={EmailMsg}
+                            labelAlign='left'
+                            style={{ textAlign: 'right' }}                                 
+                        >
+                            <Input placeholder={'name@example.org'}  style={INPUT_STYLE}/>
```
Comment (pahaz) [inline apps/condo/pages/auth/register-new.tsx:238]: it's better to extract to lang

File: apps/condo/pages/auth/signin.tsx [hunk 47]
```diff
@@ -1,170 +1,162 @@
...
+                        {SignInMsg}
+                    </Button>
+
+                    <Typography.Text  type='secondary'>
+                        <FormattedMessage
+                            id='pages.auth.signin.ResetPasswordLink'
+                            values={{
+                                link: <a style={LINK_STYLE} onClick={() => Router.push('/auth/forgot')}>{ResetMsg}</a>,
+                            }}
+                        ></FormattedMessage>
```
Comment (pahaz) [inline apps/condo/pages/auth/signin.tsx:132]:
```suggestion
                        />
```

File: apps/condo/domains/user/schema/AuthPassword.js [hunk 48]
```diff
@@ -0,0 +1,56 @@
...
+            type: 'type AuthenticateUserWithPhoneAndPasswordOutput { item: User }',
+        },
+    ],
+    mutations: [
+        {
+            access: true,
+            schema: 'authenticateUserWithPhoneAndPassword(data: AuthenticateUserWithPhoneAndPasswordInput!): AuthenticateUserWithPhoneAndPasswordOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { phone: inputPhone, password } = info.variableValues
+                const phone = inputPhone.replace(PHONE_CLEAR_REGEXP, '')
```
[the remark this answers is not in the document]
Comment (dkoviazin) [reply apps/condo/domains/user/schema/AuthPassword.js:24]: The same regexp is used at user schema on the phone field input hook. So, we need to use it here - to make the same format

File: apps/condo/public/resetEmail.html [hunk 49]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (dkoviazin) [reply apps/condo/public/resetEmail.html:1]: It's a temporary file - will move it to thew Mailgun templates later

**Pull request merged**